avm/res/event-grid/system-topic/main.bicep

```diff
@@ -85,15 +88,25 @@ var builtInRoleNames = {
 
 var formattedRoleAssignments = [
   for (roleAssignment, index) in (roleAssignments ?? []): union(roleAssignment, {
-    roleDefinitionId: builtInRoleNames[?roleAssignment.roleDefinitionIdOrName] ?? (contains(
-        roleAssignment.roleDefinitionIdOrName,
+    roleDefinitionId: builtInRoleNames[?roleAssignment.roleDefinitionId] ?? (contains(
```

Comment:
```suggestion
    roleDefinitionId: builtInRoleNames[?roleAssignment.roleDefinitionIdOrName] ?? (contains(
```

Reply:
The default role assignment implementation must remain the same - only the one for the externalRoleAssignment must be changed